services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -0,0 +1,73 @@
+package org.openkilda.functionaltests.spec.northbound.flows
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.FlowHelper
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.northbound.NorthboundService
+import org.openkilda.testing.service.topology.TopologyEngineService
+import org.springframework.beans.factory.annotation.Autowired
+
+class BandwidthSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    TopologyEngineService topologyEngineService
+    @Autowired
+    FlowHelper flowHelper
+    @Autowired
+    NorthboundService northboundService
+
+    def "Available bandwidth on ISLs changes respectively when creating a flow"() {
+        when: "Create flow with a valid bandwidth"
+        def maximumBandwidth = 1000
+        List<List<PathNode>> allPaths = []
+
+        def switches = topology.getActiveSwitches()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 1
+        }
+        srcSwitch && dstSwitch
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        def links = northboundService.getAllLinks()
+        def forwardLink = links.find() {
+            it.path.first().switchId == flow.source.datapath &&
+                    it.path.last().switchId == flow.destination.datapath
+        }
+        def reverseLink = links.find {
+            it.path.first().switchId == flow.destination.datapath &&
+                    it.path.last().switchId == flow.source.datapath
+        }
```

Comment:
The above two finds does not consider ports. two switches can have more than one link (this even was an actual case on Staging for some time)